specification/configuration.md

```diff
@@ -159,8 +159,9 @@ instance using the following environment variables:
 | Name (default value)                            | Description                                                    |
 | :------------------------------------:          | :--------------------------------------------------------:     |
 | `SPLUNK_ACCESS_TOKEN` ()                        | Access token added to exported data. [1]                       |
-| `SPLUNK_TRACE_RESPONSE_HEADER_ENABLED` (`true`) | Whether `Server-Timing` header is added to HTTP responses. [2] |
-| `SPLUNK_METRICS_ENDPOINT` ()                    | Endpoint for metrics data ingest.                     |
+| `SPLUNK_METRICS_ENDPOINT` ()                    | Endpoint for metrics data ingest.                              |
+| `SPLUNK_REALM` (`none`)                         | Which realm to send exported data. [2]                         |
```

Comment:
Other vars don't have a `none` default value and instead just default to nothing/empty string. Do we need a special default value ("none" string literal) for SPLUNK_REALM?

Reply:
I think we want to avoid using empty strings in general and use `none` in such cases in general: https://github.com/open-telemetry/opentelemetry-specification/pull/2102

It's also consistent with setting exporters, propagators (was that PR merged already?)

Reply:
That makes sense as it allows users to easily "unset" a variable that has a non-none default but in case the variable does not have a default, do we want to specify `none` as the default value or just say it has no default? I know it does not make a difference in implementation but when reading the spec it can be confusing when some values do not have a default value while others have `none` as default value while both having the same net effect.

Reply:
> Other vars don't have a none default value and instead just default to nothing/empty string.

@owais, can you give an example of an enum configuration that works like that? If you find one, then I agree that `none` value make only sense when the default value is "not none".

Reply:
> Other vars don't have a `none` default value and instead just default to nothing/empty string

@owais Can you give an example of such enum-type configuration, please?

Reply:
<img width="206" alt="image" src="https://user-images.githubusercontent.com/46186/141975101-d90444b5-a715-4068-b46d-663c87ce6e38.png">
<img width="327" alt="image" src="https://user-images.githubusercontent.com/46186/141975143-55633e3e-b3d0-43b3-94a2-f99806b6a1bc.png">
<img width="355" alt="image" src="https://user-images.githubusercontent.com/46186/141975175-70499ba7-85a4-4fd2-994b-e2ef69e356f5.png">
<img width="192" alt="image" src="https://user-images.githubusercontent.com/46186/141975225-c21e5ef2-1012-490e-93b3-44f14987ac79.png">


None of these have any default values. How do we decide when something does not have a default vs when something has default set to `none`?

Reply:
`none` is for enum value ([more](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#enum-value). I think that only the Kubernetes configs could be enums but they look more like dummy strings at the moment - moreover there are experimental.

Reply:
Yes, that makes sense and like I said this is not a hard blocker for me. Just that it can be confusing to me as an end user when I see a dozen options with some listing the default as `none` while others listing no default value and all of them behaving the same. As a consumer, I'd prefer a library to have uniform defaults/noop representations.

Feel free to resolve this if you'd like :)